python/PyQt6/core/__init__.py.in

```diff
@@ -750,6 +764,30 @@ try:
 
    QgsGeometry.as_numpy = _qgsgeometry_as_numpy
 
+   def _fields_as_numpy(self, field_names: _typing.List[str], null_value: _typing.Optional[object], request: _typing.Optional[QgsFeatureRequest] = None, target_type: _typing.Optional[_numpy.dtype] = _numpy.int32) -> _numpy.ma.MaskedArray:
+      qmeta_type = _numpy_dtype_to_qmetatype(target_type)
+    
+      field_indices = [self.fields().lookupField(name) for name in field_names]
+      for field_index in field_indices:
+          if field_index < 0:
+            raise KeyError(f"Field '{field_names[field_index]}' does not exist.")
+
+      if request is None:
+         request = QgsFeatureRequest().setFlags(Qgis.FeatureRequestFlag.NoGeometry).setSubsetOfAttributes(field_indices)
+
+      if request.limit == -1:
+         n_features = self.featureCount()
```

Comment:
featureCount() is unreliable -- it may return negative values if unknown, and may be out of sync with the actual number of features which will come from the data provider. Better to count the *actual* number of features in fieldsToDataArray and return that from the c++ method.